### DISCUSSION TITLE
Ability to download multiple/all versions of a model or different models at a time

### DISCUSSION THREAD

**Discussion by mosherc** [BlafKing/sd-civitai-browser-plus#57] [Ideas]
A lot of times, I want multiple versions or all version of a model on civit AI. It seems like download URLs are single versions only for now, so maybe a way to multi-select versions and then download them all in sequence (obviously will take a while but faster than manually selecting each version). I think it would default to the directory chosen for the whole model. As far as multiple files per version, I guess this could just download the default version, or this could be based on the existing setting of downloading all files.

Here's an example of a model split into multiple versions where I might just want all of them.

Another idea that I've been wanting is the ability to select multiple _different_ models to download at the same time, where I either ctrl+click or click a checkbox in a multi-select mode, and then select the directory (for simplicity, this should probably download the default file of the default version). Ex. would be I have a tag that I like to download models from often, and there are 5 new models in that tag, it would be nice to select all 5 for download at once similar to above.

**Comment by BlafKing** [comment]
I'd love to have this implemented as well, but, unfortunately implementing a download queue will be quite tricky, mainly because of the way that Gradio (Library used for the UI in SD-WebUI) works.
It's not possible to easily pass any info to a function that is triggered by a gradio event and is already running, each component requires a return before it can be changed/processed again.

It might be possible to implement, but it will be rather tricky to do so.
Not sure if my Python skills are sufficient for it :,)

I'll definitely look into though because I'd love to have this feature as well! :) (No promises though 😅)

**Reply by mosherc** [reply]
Hey that's good enough for me! Thanks

**Reply by SchofieldPriest** [reply]
Leaving a comment here because I would love to see a "Download all results at once" feature implemented at some point! With an option to specify a download folder and specify the exact results with quotation marks (a separate idea I posted), this feature would make multi-downloads an absolute breeze.

**Comment by metalgie** [comment]
This feature should be there, I love your extension, but download queue would make it a masterpiece

**Comment by BlafKing** [comment]
Good news!
I finally got around to getting the basics down for implementing a functional download queue and it's looking to be working well so far.

I'll now start working on expanding the functionality/features so you can expect an update in about 1-2 days with a functioning download queue! :)

**Reply by metalgie** [reply]
amazing, can't wait for this new feature :D

**Comment by BlafKing** [comment]
I've now published an update with a (hopefully) functional download queue! :) [(v3.0)](https://github.com/BlafKing/sd-civitai-browser-plus/commit/fecf7900ab7fddc4da5578f1e0e8eb2db5271b94)
Took a lot of head scratching but it should all be working now!

It's not yet possible to select multiple versions since I couldn't find an easy way to implement this, so it'll take a bit more time to figure that out.

I'll close this thread now since the request is (partially) fulfilled.
Please do feel free to still use this thread to give more suggestions about the download queue though!
